Quickstart: 2 of 3 contracts fail to compile out of the box + missing constructor args

## Summary

Following the [quickstart guide](https://scaffoldstellar.org/docs/quick-start) from a fresh `stellar scaffold init`, `npm run dev` fails to build 2 of 3 example contracts (`fungible-allowlist` and `nft-enumerable`). After fixing the compile errors, client generation also fails for those two contracts due to missing constructor arguments in `environments.toml`.

The `guess-the-number` contract works end-to-end.

## Environment

- **Stellar CLI:** v25.1.0
- **Rust:** 1.89.0 (from `rust-toolchain.toml`)
- **soroban-sdk:** 23.1.0
- **OpenZeppelin stellar-contracts:** v0.5.1 (git tag)
- **Node:** v24.13.0
- **OS:** macOS (Darwin 25.3.0, arm64)

## Bug 1: `#[contractimpl(contracttrait)]` not recognized by soroban-sdk 23.1.0

The `fungible-allowlist` and `nft-enumerable` contract templates use `#[contractimpl(contracttrait)]` on trait impl blocks:

```rust
#[contractimpl(contracttrait)]
impl FungibleToken for ExampleContract {
    type ContractType = AllowList;
}
```

This produces:

```
error: Unknown field: `contracttrait`
  --> contracts/fungible-allowlist/src/contract.rs:47:16
   |
47 | #[contractimpl(contracttrait)]
   |                ^^^^^^^^^^^^^
```

`soroban-sdk 23.1.0` does not support a `contracttrait` parameter on `#[contractimpl]`. This affects 6 locations across both contracts (lines 47, 68, 71 in `fungible-allowlist`; lines 37, 42, 45 in `nft-enumerable`) and causes 12 total errors (the rest are cascading trait-bound failures).

**Likely cause:** The templates appear to target OpenZeppelin stellar-contracts v0.6.0 (see #370), but `Cargo.toml` pins `tag = "v0.5.1"`. The `contracttrait` syntax may be a v0.6.0 feature that doesn't exist yet in v0.5.1/soroban-sdk 23.1.0.

**Workaround:** Replace `#[contractimpl(contracttrait)]` with `#[default_impl]` + `#[contractimpl]` (using `stellar_macros::default_impl`):

```rust
use stellar_macros::default_impl;

#[default_impl]
#[contractimpl]
impl FungibleToken for ExampleContract {
    type ContractType = AllowList;
}
```

## Bug 2: `grant_role_no_auth` arguments swapped in `fungible-allowlist`

In `contracts/fungible-allowlist/src/contract.rs:37`:

```rust
// Current (broken) — account and role args are swapped:
access_control::grant_role_no_auth(e, &manager, &symbol_short!("manager"), &admin);

// Fixed — matches the function signature (caller, account, role):
access_control::grant_role_no_auth(e, &manager, &admin, &symbol_short!("manager"));
```

Error:
```
error[E0308]: mismatched types
  --> contracts/fungible-allowlist/src/contract.rs:37:58
   |
37 |         access_control::grant_role_no_auth(e, &manager, &symbol_short!("manager"), &admin);
   |                                                          ^^^^^^^^^^^^^^^^^^^^^^^^ expected `Address`, found `Symbol`
```

## Bug 3: Missing constructor args in `environments.toml`

After fixing the compile errors, all 3 contracts build and deploy, but client generation fails for two:

```
❌ Failed to generate client for: fungible_allowlist_example
   Missing required argument 'name' of type string

❌ Failed to generate client for: nft_enumerable_example
   Missing required argument 'uri' of type string
```

The `constructor_args` in `environments.toml` are incomplete:

| Contract | Current `constructor_args` | Missing |
|---|---|---|
| `fungible_allowlist_example` | `--admin me --manager me --initial_supply ...` | `--name` and `--symbol` |
| `nft_enumerable_example` | `--owner me` | `--uri`, `--name`, and `--symbol` |

**Fix:**
```toml
fungible_allowlist_example = { client = true, constructor_args = "--name ExampleToken --symbol EXT --admin me --manager me --initial_supply 1000000000000000000000000" }
nft_enumerable_example = { client = true, constructor_args = "--uri https://example.com/nft/ --name ExampleNFT --symbol ENFT --owner me" }
```

## Steps to Reproduce

1. `stellar scaffold init my-project`
2. `cd my-project && npm install`
3. `npm run dev`
4. Observe compile errors for `fungible-allowlist` and `nft-enumerable`

## Related

- #370 — Update scaffold dependencies to be compatible with OpenZeppelin contracts @ v0.6.0

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,19 +10,19 @@ repository = "https://github.com/theahaco/scaffold-stellar"
 version = "0.0.1"
 
 [workspace.dependencies.soroban-sdk]
-version = "23.1.0"
+version = "23.4.0"
 
 [workspace.dependencies.stellar-access]
 git = "https://github.com/OpenZeppelin/stellar-contracts"
-tag = "v0.5.1"
+tag = "v0.6.0"
 
 [workspace.dependencies.stellar-macros]
 git = "https://github.com/OpenZeppelin/stellar-contracts"
-tag = "v0.5.1"
+tag = "v0.6.0"
 
 [workspace.dependencies.stellar-tokens]
 git = "https://github.com/OpenZeppelin/stellar-contracts"
-tag = "v0.5.1"
+tag = "v0.6.0"
 
 [profile.release]
 opt-level = "z"

--- a/contracts/fungible-allowlist/src/contract.rs
+++ b/contracts/fungible-allowlist/src/contract.rs
@@ -5,9 +5,11 @@
 //! controlled token transfers by an admin who can allow or disallow specific
 //! accounts.
 
-use soroban_sdk::{contract, contractimpl, symbol_short, Address, Env, String};
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, Address, Env, MuxedAddress, String, Symbol, Vec,
+};
 use stellar_access::access_control::{self as access_control, AccessControl};
-use stellar_macros::{default_impl, only_role};
+use stellar_macros::only_role;
 use stellar_tokens::fungible::{
     allowlist::{AllowList, FungibleAllowList},
     burnable::FungibleBurnable,
@@ -19,18 +21,20 @@ pub struct ExampleContract;
 
 #[contractimpl]
 impl ExampleContract {
-    pub fn __constructor(e: &Env, admin: Address, manager: Address, initial_supply: i128) {
-        Base::set_metadata(
-            e,
-            18,
-            String::from_str(e, "AllowList Token"),
-            String::from_str(e, "ALT"),
-        );
+    pub fn __constructor(
+        e: &Env,
+        name: String,
+        symbol: String,
+        admin: Address,
+        manager: Address,
+        initial_supply: i128,
+    ) {
+        Base::set_metadata(e, 18, name, symbol);
 
         access_control::set_admin(e, &admin);
 
         // create a role "manager" and grant it to `manager`
-        access_control::grant_role_no_auth(e, &admin, &manager, &symbol_short!("manager"));
+        access_control::grant_role_no_auth(e, &manager, &admin, &symbol_short!("manager"));
 
         // Allow the admin to transfer tokens
         AllowList::allow_user(e, &admin);
@@ -40,8 +44,7 @@ impl ExampleContract {
     }
 }
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl FungibleToken for ExampleContract {
     type ContractType = AllowList;
 }
@@ -62,10 +65,8 @@ impl FungibleAllowList for ExampleContract {
     }
 }
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl AccessControl for ExampleContract {}
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl FungibleBurnable for ExampleContract {}

--- a/contracts/fungible-allowlist/src/contract.rs
+++ b/contracts/fungible-allowlist/src/contract.rs
@@ -34,7 +34,7 @@ impl ExampleContract {
         access_control::set_admin(e, &admin);
 
         // create a role "manager" and grant it to `manager`
-        access_control::grant_role_no_auth(e, &manager, &admin, &symbol_short!("manager"));
+        access_control::grant_role_no_auth(e, &manager, &symbol_short!("manager"), &admin);
 
         // Allow the admin to transfer tokens
         AllowList::allow_user(e, &admin);

--- a/contracts/nft-enumerable/src/contract.rs
+++ b/contracts/nft-enumerable/src/contract.rs
@@ -5,7 +5,6 @@
 //! IDs owned by each account.
 
 use soroban_sdk::{contract, contractimpl, contracttype, Address, Env, String};
-use stellar_macros::default_impl;
 use stellar_tokens::non_fungible::{
     burnable::NonFungibleBurnable,
     enumerable::{Enumerable, NonFungibleEnumerable},
@@ -22,14 +21,9 @@ pub struct ExampleContract;
 
 #[contractimpl]
 impl ExampleContract {
-    pub fn __constructor(e: &Env, owner: Address) {
+    pub fn __constructor(e: &Env, uri: String, name: String, symbol: String, owner: Address) {
         e.storage().instance().set(&DataKey::Owner, &owner);
-        Base::set_metadata(
-            e,
-            String::from_str(e, "www.mytoken.com"),
-            String::from_str(e, "My Token"),
-            String::from_str(e, "TKN"),
-        );
+        Base::set_metadata(e, uri, name, symbol);
     }
 
     pub fn mint(e: &Env, to: Address) -> u32 {
@@ -40,16 +34,13 @@ impl ExampleContract {
     }
 }
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl NonFungibleToken for ExampleContract {
     type ContractType = Enumerable;
 }
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl NonFungibleEnumerable for ExampleContract {}
 
-#[default_impl]
-#[contractimpl]
+#[contractimpl(contracttrait)]
 impl NonFungibleBurnable for ExampleContract {}

--- a/environments.toml
+++ b/environments.toml
@@ -9,8 +9,8 @@ name = "me" # Required. Keys for this account will be saved to `./.stellar/ident
 default = true # Optional. Whether to use this account as the `--source` for commands that need one.
 
 [development.contracts]
-fungible_allowlist_example = { client = true, constructor_args = "--admin me --manager me --initial_supply 1000000000000000000000000" }
-nft_enumerable_example = { client = true, constructor_args = "--owner me" }
+fungible_allowlist_example = { client = true, constructor_args = "--name ExampleToken --symbol EXT --admin me --manager me --initial_supply 1000000000000000000000000" }
+nft_enumerable_example = { client = true, constructor_args = "--uri https://example.com/nft/ --name ExampleNFT --symbol ENFT --owner me" }
 
 # Rather than in one list, TOML allows specifying contracts in their own "sections"
 [development.contracts.guess_the_number]

--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
     "dev": "npm start",
     "start": "concurrently --kill-others-on-fail --names stellar,vite -c gray,green --pad-prefix \"stellar scaffold watch --build-clients\" \"vite\"",
     "build": "tsc -b && vite build",
-    "install:contracts": "npm install --workspace=packages && npm run build --workspace=packages",
+    "install:contracts": "npm install --workspaces && npm run build --workspaces",
     "preview": "vite preview",
     "lint": "eslint .",
     "format": "prettier . --write",

